runners/samza/src/test/java/org/apache/beam/runners/samza/translation/TranslationContextTest.java

```diff
@@ -43,45 +53,104 @@
-            inputStream.partitionBy(keyFn, valueFn, serde, streamName);
-          },
-          getConfig());
-  Map<PValue, String> idMap = new HashMap<>();
-  Set<String> nonUniqueStateIds = new HashSet<>();
-  TranslationContext translationContext =
-      new TranslationContext(
-          streamApplicationDescriptor, idMap, nonUniqueStateIds, mock(SamzaPipelineOptions.class));
+  TranslationContext translationContext;
+  AppliedPTransform pTransform;
+  PCollection output;
+  List inputDescriptors;
+  StreamApplicationDescriptor streamApplicationDescriptor;
+  SamzaPipelineOptions pipelineOptions;
+  TransformHierarchy.Node node;
 
-  @Test
-  public void testRegisterInputMessageStreams() {
-    final PCollection output = mock(PCollection.class);
+  @Before
+  public void before() {
+    final GenericInputDescriptor testInputDescriptor =
+        new GenericSystemDescriptor("mockSystem", "mockFactoryClassName")
+            .getInputDescriptor("test-input-1", mock(Serde.class));
+    final MapFunction<Object, String> keyFn = m -> m.toString();
+    final MapFunction<Object, Object> valueFn = m -> m;
+    final String streamName = "testStream";
+    KVSerde<Object, Object> serde = KVSerde.of(new NoOpSerde<>(), new NoOpSerde<>());
+    pipelineOptions = mock(SamzaPipelineOptions.class);
+    when(pipelineOptions.getEnableTransformMetrics()).thenReturn(false);
+    // Create a stream application descriptor with a partitionBy
+    streamApplicationDescriptor =
+        new StreamApplicationDescriptorImpl(
+            appDesc -> {
+              MessageStream inputStream = appDesc.getInputStream(testInputDescriptor);
+              inputStream.partitionBy(keyFn, valueFn, serde, streamName);
+            },
+            getConfig());
+    Map<PValue, String> idMap = new HashMap<>();
+    Set<String> nonUniqueStateIds = new HashSet<>();
+    translationContext =
+        new TranslationContext(
+            streamApplicationDescriptor, idMap, nonUniqueStateIds, pipelineOptions);
+
+    // Register the input message stream
+    output = mock(PCollection.class);
+    pTransform = mock(AppliedPTransform.class);
+    node = mock(TransformHierarchy.Node.class);
+    when(pTransform.getFullName()).thenReturn("mock-ptransform");
     List<String> topics = Arrays.asList("stream1", "stream2");
-    List inputDescriptors =
+    inputDescriptors =
         topics.stream()
             .map(topicName -> createSamzaInputDescriptor(topicName, topicName))
             .collect(Collectors.toList());
+  }
 
+  @Test
+  public void testRegisterInputMessageStreams() {
+    translationContext.setCurrentTransform(pTransform);
     translationContext.registerInputMessageStreams(output, inputDescriptors);
-
     assertNotNull(translationContext.getMessageStream(output));
   }
 
-  public GenericInputDescriptor<KV<String, OpMessage<?>>> createSamzaInputDescriptor(
+  @Test
+  public void testMetricOpNotAttachedForIOTransform() {
```

Comment:
Let's remove testMetricOpNotAttachedForIOTransform() and testMetricOpNotAttachedForConfigOveride() with the usage of powermock all together. These tests do not add much value, and brings the heavy powermock into the tests. Let's remove them.

Reply:
done